src/app/debug/filter-side-drawer/filter.service.ts

```diff
@@ -0,0 +1,35 @@
+import { Injectable } from '@angular/core';
+import { Observable, Subject } from 'rxjs';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class FilterService {
+  private showFilterSubject: Subject<boolean> = new Subject();
+  showFilterObserver: Observable<boolean> = this.showFilterSubject.asObservable();
+
+  private metadataNamesSubject: Subject<string[]> = new Subject();
+  metadataNamesObserver: Observable<string[]> = this.metadataNamesSubject.asObservable();
+
+  private filters: Record<string, string> = {};
+  private filterContextSubject: Subject<Record<string, string>> = new Subject();
+  filterContextObserver: Observable<Record<string, string>> = this.filterContextSubject.asObservable();
```

Comment:
Here to. Might even add a $ to the observers, IDK what @Matthbo thinks about that. I'll add it to the conventions.

Reply:
In RxJS and Angular docs its common to indicate a subject by either a normal variable name or with the suffix "Subject".
Then the Observable version of that subject will have the normal variable name with the suffix "$" instead.
```ts
const booleanVarSubject: Subject<boolean> = new Subject(false);
const booleanVar$: Observable<boolean> = booleanVarSubject.asObservable();
```
To me this seems very clear since the subject can do everything while the observable is only meant for public accessibility and can only be subscribed to